tools/changelog.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Grab the changelog for issues."""
+
+import argparse
+import datetime
+import os
+import sys
+from urllib import quote_plus
+from urlparse import urlunsplit
+
+import requests
+
+# This will start an initialization process.
+sys.path.append(os.path.realpath(os.getcwd()))
+from config.secrets import OAUTH_TOKEN
+
+# Config
+GITHUB_API = 'api.github.com'
+ROOT_REPO = '/repos/webcompat'
+REPO_TEST = 'webcompat-tests'
+REPO_BUGS = 'webcompat.com'
+QUERY_PARAM = 'labels'
+QUERY_VALUE = 'status: add to changelog'
+HEADERS = {'Accept': 'application/vnd.github.v3+json',
+           'Authorization': 'token {token}'.format(token=OAUTH_TOKEN),
+           'User-Agent': 'webcompat/webcompat-bot'}
+# Templates for producing the changelog
+LINE_TEMPLATE = '* {title} [Pull #{number}]({url})\n'
+LOG_TEMPLATE = """
+
+X.X.X - {date}
+{loglines}
+
+"""
+
+
+def get_remote_file(url):
+    """Request URL."""
+    r = requests.get(url, headers=HEADERS)
+    json_response = r.json()
+    return json_response
+
+
+def create_changelog(json_response):
+    """Extract the number of open issues."""
+    loglines = ''
+    for issue in json_response:
+        title = issue['title']
+        number = issue['number']
+        url = issue['url']
+        loglines += LINE_TEMPLATE.format(title=title, url=url, number=number)
+    return loglines
+
+
+def delete_label(url):
+    """Delete the label for an issue."""
+    r = requests.delete(url, headers=HEADERS)
+    return r.status_code
+
+
+def main():
+    """Core program."""
+    # Parsing arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--delete', action='store_true',
+                        help='delete the changelog label on GitHub.')
+    parser.add_argument('-t', '--testmode', action='store_true',
+                        help='try on webcompat-tests only.')
+    args = parser.parse_args()
+
+    # Impossible to erase things by default.
+    erase_label = False
+
+    if args.delete:
+        erase_label = True
+    if args.testmode:
+        repo = REPO_TEST
+    else:
+        repo = REPO_BUGS
+
+    # Create the query PATH
+    path = '{root}/{repo}/issues'.format(root=ROOT_REPO, repo=repo)
+    query = '{param}={value}'.format(
```

Comment:
We also need to add `state=closed`, otherwise we get no results (because the default is `state=open`). 

This works in test mode, because the test issue with the label is open. 

(or we could add `state=all` in case we think we might miss PRs that weren't closed by accident).

Reply:
(locally I just did `query = 'state=all&{param}={value}'.format(` and it worked like a charm for the webcompat.com dev repo.

Reply:
done